api/src/internal/console.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export type ConsoleLogFunction = (message: string, ...args: unknown[]) => void;
+
+/**
+ * Internal API to encapsulate console related calls to manage them from a single point.
+ *
+ * @since 1.9.1
+ */
+export interface ConsoleConfig {
+  log?: ConsoleLogFunction;
+  error?: ConsoleLogFunction;
+  warn?: ConsoleLogFunction;
+  info?: ConsoleLogFunction;
+  debug?: ConsoleLogFunction;
+  trace?: ConsoleLogFunction;
+  dir?: (item?: any, options?: any) => void;
+}
+
+export type ConsoleFunctionNames =
+  | 'log'
+  | 'error'
+  | 'warn'
+  | 'info'
+  | 'debug'
+  | 'trace'
+  | 'dir';
+
+let activeConfig: ConsoleConfig = {};
+let scopeObj: any;
```

Comment:
Not clear what this is for

Reply:
it is the configured object to be used while calling console function with the `apply`